### PR TITLE
Various fixes

### LINKED ISSUE #309: Looting Items from the ground is not displaying properly
Issue:
If two or more people are in a map and another user loots an item, it'll display as if you looted the item. 
You don't get the item, the player who looted gets it.

### LINKED ISSUE #315: Event map portal
Leaving the event map through the exit portal doesn't seem to be working

### PR DESCRIPTION
**Pull request by AngeloTadeucci**
- Fix #309 
- Fix #315 
- Fixed home templates not working correctly;
- Fixed newly created homes not having the correct instance;
- Fixed second rider of two person mounts not leaving the mount correctly;
- Beauty Salon/Dye Workshop/Plastic surgery are now instanced and the return portal works.

### CODE REVIEW

File: MapleServer2/PacketHandlers/Game/BeautyHandler.cs [hunk 1]
```diff
@@ -360,7 +360,9 @@ private static void HandleTeleport(GameSession session, PacketReader packet)
                     return;
             }
 
-            MoveFieldHandler.HandleInstanceMove(session, (int) mapId);
+            session.Player.ReturnCoord = session.FieldPlayer.Coord;
+            session.Player.ReturnMapId = session.Player.MapId;
+            session.Player.Warp((int) mapId, instanceId: session.Player.CharacterId);
```
Comment (Sparkymod) [inline MapleServer2/PacketHandlers/Game/BeautyHandler.cs:365]: this is confusing, use one format or the other, `mapId: (int) mapId, instanceId: session.Player.CharacterId`

File: MapleServer2/PacketHandlers/Game/RequestCubeHandler.cs [hunk 2]
```diff
@@ -174,7 +174,7 @@ private static void HandleLoadFurnishingItem(GameSession session, PacketReader p
         private static void HandleBuyPlot(GameSession session, PacketReader packet)
         {
             int groupId = packet.ReadInt();
-            int homeTemplate = packet.ReadInt();
+            int homeTemplate = packet.ReadInt() + 1;
```
Comment (Sparkymod) [inline MapleServer2/PacketHandlers/Game/RequestCubeHandler.cs:177]: for what is this ?

**Review by Sparkymod — CHANGES_REQUESTED** [review]

File: MapleServer2/PacketHandlers/Game/RequestCubeHandler.cs (hunk 2 above)
Comment (AngeloTadeucci) [reply MapleServer2/PacketHandlers/Game/RequestCubeHandler.cs:177]: The client requests home templates from 0~2, the XMLs ids are from 1~3. Should I add an comment or maybe move the +1 to inside the `Home.SetTemplate` function?

Comment (Sparkymod) [reply MapleServer2/PacketHandlers/Game/RequestCubeHandler.cs:177]: a comment should be good tho, `// the Templates ids in the XMLs are from 1-3 and the client request 0-2`

Comment (AngeloTadeucci) [reply MapleServer2/PacketHandlers/Game/RequestCubeHandler.cs]: I changed, at least the +1 isn't on the packet.ReadInt(). It'll stay inside the Home() constructor

**Review by Sparkymod — APPROVED** [review]

File: MapleServer2/Tools/HomeManager.cs [hunk 3]
```diff
@@ -20,7 +20,12 @@ public HomeManager()
             SaveLoop();
         }
 
-        public void AddHome(Home home) => HomeList[home.Id] = home;
+        public void AddHome(Home home)
+        {
+            home.InstanceId = InstanceCounter;
+            HomeList[home.Id] = home;
+            IncrementCounter();
```
Comment (AlanMorel) [inline MapleServer2/Tools/HomeManager.cs:27]: Can we just make `IncrementCounter()` return the current count so we can just set it directly all in one motion? Like `home.InstanceId = IncrementCounter();`

**Review by AlanMorel — CHANGES_REQUESTED** [review]

**Review by AlanMorel — APPROVED** [review]

**Pull request merged**